spec/unit/file_serving/http_metadata_spec.rb

```diff
@@ -29,6 +29,13 @@
       expect( metadata.mode ).to be_nil
     end
 
+    it "skips md5 checksum type in collect on FIPS enabled platforms" do
+      allow(Puppet::Util::Platform).to receive(:fips_enabled?).and_return(true)
+      metadata = described_class.new(http_response)
+      metadata.collect
+      expect( metadata.checksum_type ).to_not eq :md5
```

Comment:
I think you'll need to add an `X-Checksum-Md5` header to the `http_response` to test that the `collect` method falls back to `mtime`, something like

```ruby
allow(http_response).to receive(:[]).with('X-Checksum-Md5').and_return(<some valid MD5 checksum>)
```

Also recommend having a "positive" expectation like `expect( metadata.checksum_type ).to eq :mtime` otherwise the test might pass if `checksum_type` returned `nil`